File: streaming-s3-example-java/src/main/java/nl/vintik/streaming/java/FileRequest.java
package nl.vintik.streaming.java;

/**
 * Typed request for a file-serving endpoint: carries the validated file name that is
 * used to address the object in S3.
 *
 * <p>Java counterpart of the Kotlin example's {@code FileRequest}. Consumed by
 * {@code S3Source} when confirming existence/size and opening the object body.
 *
 * @param fileName the validated S3 object key requested by the client
 */
public record FileRequest(String fileName) {}

File: streaming-s3-example/src/main/kotlin/nl/vintik/lambda/streaming/FileRequest.kt
package nl.vintik.lambda.streaming

/**
 * Typed request for a file-serving endpoint: carries the validated file name extracted
 * from the API Gateway `/{proxy+}` path parameter.
 *
 * Produced by [FileKeyResolver] and consumed by [S3Source]. Extend this data class (or
 * create a richer variant) if your use case requires additional fields such as tenant ID,
 * region, or access scope.
 */
data class FileRequest(val fileName: String)
